core/admin/admin.php

```diff
@@ -701,6 +701,12 @@ public function admin_action_new_post() {
 
 		$post_data = Utils::get_super_global_value( $_GET, 'post_data' ) ?? [];
 
+		if ( ! current_user_can( 'publish_posts' ) && isset( $post_data['post_status'] ) && 'publish' === $post_data['post_status'] ) {
+			$post_data['post_status'] = 'draft';
```

Comment:
@mugurelLupu What is the purpose of setting a `post_status` here, if you will just remove again it below? `wp_insert_post()` will make it  draft automatically if post_status is not specified. Can you please double check this aspect? Thanks!